File: src/main/java/net/insprill/cjm/listeners/QuitEvent.java
package net.insprill.cjm.listeners;

import net.insprill.cjm.CustomJoinMessages;
import net.insprill.cjm.handlers.PlayerHandler;
import net.insprill.cjm.messages.MessageAction;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerQuitEvent;

public class QuitEvent implements Listener {

    private final CustomJoinMessages plugin;

    public QuitEvent(CustomJoinMessages plugin) {
        this.plugin = plugin;
    }

    @EventHandler(priority = EventPriority.MONITOR)
    public void onPlayerQuit(PlayerQuitEvent e) {
        e.setQuitMessage("");
        plugin.getMessageSender().sendMessages(e.getPlayer(), MessageAction.QUIT, true);
        PlayerHandler.setStatus(e.getPlayer().getUniqueId(), PlayerHandler.Status.LOGGED_OUT);
    }

}

File: src/main/kotlin/net/insprill/cjm/listener/QuitEvent.kt
package net.insprill.cjm.listener

import net.insprill.cjm.CustomJoinMessages
import net.insprill.cjm.message.MessageAction
import org.bukkit.event.EventHandler
import org.bukkit.event.EventPriority
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent

class QuitEvent(private val plugin: CustomJoinMessages) : Listener {

    @EventHandler(priority = EventPriority.LOWEST)
    fun onPlayerQuit(e: PlayerQuitEvent) {
        e.quitMessage = null
        plugin.messageSender.trySendMessages(e.player, MessageAction.QUIT, true)
    }

}
